Detections/MultipleDataSources/MailBoxTampering.yaml

```diff
@@ -0,0 +1,79 @@
+    dataTypes:
+      - SecurityAlert (ASC)
+  - connectorId: Office365
+    dataTypes:
+      - OfficeActivity
+  - connectorId: AzureActivity
+    dataTypes:
+      - AzureActivity
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1078
+  - T1548
+query: |
+  SecurityAlert 
+  | where AlertName == "Impossible travel activity"
+  | extend Extprop = parsejson(Entities)
+  | mv-expand Extprop
+  | extend Extprop = parsejson(Extprop)
+  | extend CmdLine = iff(Extprop['Type']=="process", Extprop['CommandLine'], '')
+  | extend File = iff(Extprop['Type']=="file", Extprop['Name'], '')
+  | extend Account = Extprop['Name']
+  | extend Domain = Extprop['UPNSuffix']
+  | extend Account = iif(isnotempty(Domain) and Extprop['Type']=="account", tolower(strcat(Account, "@", Domain)), iif(Extprop['Type']=="account", tolower(Account), ""))
+  | extend IpAddress = iff(Extprop["Type"] == "ip",Extprop['Address'], '')
+  | extend Process = iff(isnotempty(CmdLine), CmdLine, File)
+  | project TimeGenerated,Account,IpAddress,CompromisedEntity,Description,ProviderName,ResourceId
+  | join kind=inner
+  (
+  OfficeActivity
+  | where Operation =~ "Add-MailboxPermission"
+  | extend value = tostring(parse_json(Parameters)[3].Value)
+  | where value contains "FullAccess"
+  | where ResultStatus == "True"
+  | project Parameters,TimeGenerated,value,RecordType,Operation,OrganizationId,UserType,UserKey,OfficeWorkload,ResultStatus,OfficeObjectId,UserId,ClientIP,ExternalAccess,OriginatingServer,OrganizationName,TenantId,ElevationTime,SourceSystem,OfficeId,OfficeTenantId,Type,SourceRecordId
+  ) on $left.Account == $right.UserId
+  | join kind=inner
+  (
+  AuditLogs
+  | where ActivityDisplayName =~ "Add eligible member to role in PIM requested (timebound)"
+  | where AADOperationType =~ "CreateRequestEligibleRole"
+  | where TargetResources has_any ("-PRIV", "Administrator", "Security")
+  | extend BuiltinRole = tostring(parse_json(TargetResources[0].displayName))
+  | extend CustomGroup = tostring(parse_json(TargetResources[3].displayName))
+  | extend TargetAccount = tostring(parse_json(TargetResources[2].displayName))
+  | extend Initiatedby = Identity
+  | project TimeGenerated, ActivityDisplayName, AADOperationType, Initiatedby, TargetAccount, BuiltinRole, CustomGroup, LoggedByService, Result, ResourceId, Id
+  | sort by TimeGenerated desc
+  ) on $left.UserId == $right.Initiatedby
+  | project AADOperationType, ActivityDisplayName,AccountCustomEntity=Initiatedby, Id,ResourceId,IPCustomEntity=IpAddress
+
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: FullName
+        columnName: AccountCustomEntity
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IPCustomEntity
+version: 1.0.0
+kind: Scheduled
+
```

Comment:
Query looks good. Please remove the trailing space.

Reply:
> I am just curious, is this the only AuditLogs operation, related to roles, that could be triggered?
> 
> `"Add eligible member to role in PIM requested (timebound)"`

For this iteration we were only focusing on the PIM requested activity but we will look into expanding it to cover other roles as well.